src/main/kotlin/com/terraformation/backend/tracking/db/DraftPlantingSiteStore.kt

```diff
@@ -0,0 +1,118 @@
+package com.terraformation.backend.tracking.db
+
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.db.ParentStore
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.ProjectInDifferentOrganizationException
+import com.terraformation.backend.db.default_schema.OrganizationId
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.tracking.DraftPlantingSiteId
+import com.terraformation.backend.db.tracking.tables.records.DraftPlantingSitesRecord
+import com.terraformation.backend.db.tracking.tables.references.DRAFT_PLANTING_SITES
+import jakarta.inject.Named
+import java.time.InstantSource
+import java.time.ZoneId
+import org.jooq.DSLContext
+import org.jooq.JSONB
+import org.springframework.security.access.AccessDeniedException
+
+@Named
+class DraftPlantingSiteStore(
+    private val clock: InstantSource,
+    private val dslContext: DSLContext,
+    private val parentStore: ParentStore,
+) {
+  fun fetchOneById(draftPlantingSiteId: DraftPlantingSiteId): DraftPlantingSitesRecord {
+    requirePermissions { readDraftPlantingSite(draftPlantingSiteId) }
+
+    return dslContext
+        .selectFrom(DRAFT_PLANTING_SITES)
+        .where(DRAFT_PLANTING_SITES.ID.eq(draftPlantingSiteId))
+        .fetchOne()
+        ?: throw DraftPlantingSiteNotFoundException(draftPlantingSiteId)
+  }
+
+  fun create(
+      data: JSONB,
+      name: String,
+      organizationId: OrganizationId,
+      description: String? = null,
+      numPlantingSubzones: Int? = null,
+      numPlantingZones: Int? = null,
+      projectId: ProjectId? = null,
+      timeZone: ZoneId? = null,
+  ): DraftPlantingSitesRecord {
+    requirePermissions { createDraftPlantingSite(organizationId) }
+
+    if (projectId != null) {
+      requirePermissions { readProject(projectId) }
+      if (parentStore.getOrganizationId(projectId) != organizationId) {
+        throw ProjectInDifferentOrganizationException()
+      }
+    }
+
+    val now = clock.instant()
+    val userId = currentUser().userId
+
+    val record =
+        DraftPlantingSitesRecord(
+            createdBy = userId,
+            createdTime = now,
+            data = data,
+            description = description,
+            modifiedBy = userId,
+            modifiedTime = now,
+            name = name,
+            numPlantingSubzones = numPlantingSubzones,
+            numPlantingZones = numPlantingZones,
+            organizationId = organizationId,
+            projectId = projectId,
+            timeZone = timeZone,
+        )
+
+    dslContext.attach(record)
```

Comment:
just curious, is this only needed when creating new records?

Reply:
Yeah. I'm kind of trying this pattern on for size. The `DAO` classes and the POJOs (the classes that have the `Row` suffix on their names in our setup) are kind of second-class citizens in jOOQ and the official recommendation is now to use `Record` classes for most purposes instead. They use a pattern more like ActiveRecord where you tell them to persist themselves using a persistence configuration (which is what's being set by the `attach` method).